electron/renderer/static/webview-preload.js

```diff
@@ -157,19 +160,39 @@ function replaceGoogleAuth() {
 }
 
 function enableFileLogging() {
-  // TODO: add log identifier
+  // webapp uses winston ref to define log level
   global.winston = require('winston');
-  const logFile = require('../../js/config').CONSOLE_LOG
-  const logFilePath = path.join(app.getPath('userData'), 'logs', logFile);
-  console.log('Logging into file', logFilePath);
-
-  winston
-    .add(winston.transports.File, {
-      filename: logFilePath,
-      handleExceptions: true,
-    })
-    .remove(winston.transports.Console)
-    .info(pkg.productName, 'Version', pkg.version);
+
+  const id = new URL(window.location).searchParams.get('id')
+  const logName = require('../../js/config').CONSOLE_LOG
+  const logDirectory = path.join(app.getPath('userData'), 'logs');
+
+  try {
+    if (!fs.existsSync(logDirectory)) {
+      fs.mkdirSync(logDirectory);
+    }
+
+    const subDirectory = path.join(logDirectory, id);
+
+    if (!fs.existsSync(subDirectory)) {
+      fs.mkdirSync(subDirectory);
+    }
+
+    const logFilePath = path.join(subDirectory, logName);
+
+    console.log('Logging into file', logFilePath);
+
+    winston
+      .add(winston.transports.File, {
+        filename: logFilePath,
+        handleExceptions: true,
+      })
+      .remove(winston.transports.Console)
+      .info(pkg.productName, 'Version', pkg.version);
+
+  } catch (e) {
+    console.error(`Failed to create log file: ${error.message}`)
```

Comment:
This will fail because in you defined `catch (e)`. It should be `catch (error)`.